pipeline/splicing/test_checkVarHGVS.py

```diff
@@ -0,0 +1,289 @@
+
+        self.sequence = "AGT"
+
+    def test_getRevComp(self):
+        '''Test funciton getRevComp'''
+        # checks that getRevComp works for single nucleotide
+        self.sequence = "A"
+        revComp = checkVarHGVS.getRevComp(self.sequence)
+        self.assertEquals(revComp, "T")
+
+        # checks that getRevComp works for multiple nucleotides
+        self.sequence = "AGT"
+        revComp = checkVarHGVS.getRevComp(self.sequence)
+        self.assertEquals(revComp, "ACT")
+
+        # checks that getRevComp works for multiple nucleotides with unspectiifed nucleotides
+        self.sequence = "AGTNT"
+        revComp = checkVarHGVS.getRevComp(self.sequence)
+        self.assertEquals(revComp, "ANACT") 
+
+    def test_getVarInfo(self):
+        '''Tests function getVarInfo'''
+        # checks that variant is being parsed correctly
+        varInfo = checkVarHGVS.getVarInfo(self.varBRCA1)
+        self.assertEquals(varInfo["varcDNAHGVS"], self.varBRCA1["pyhgvs_cDNA"])
+        self.assertEquals(varInfo["varGenRef"], self.varBRCA1["Ref"])
+        self.assertEquals(varInfo["varGenAlt"], self.varBRCA1["Alt"])
+        self.assertEquals(varInfo["varGene"], self.varBRCA1["Gene_Symbol"])
+
+        # checks that cDNARef and cDNAAlt are set correctly for BRCA1 (reverse complement)
+        self.varBRCA1["Gene_Symbol"] = "BRCA1"
+        varInfo = checkVarHGVS.getVarInfo(self.varBRCA1)
+        self.assertEquals(varInfo["cDNARef"],"C")
+        self.assertEquals(varInfo["cDNAAlt"],"T")
+
+        # checks that cDNARef and cDNAAlt are set correctly for BRCA2
+        self.varBRCA2["Gene_Symbol"] = "BRCA2"
+        varInfo = checkVarHGVS.getVarInfo(self.varBRCA2)
+        self.assertEquals(varInfo["cDNARef"],"A")
+        self.assertEquals(varInfo["cDNAAlt"],"T")
+
+    def test_checkVarHGVSSub(self):
+        '''Tests checkVarHGVS for single nucleotide substitutions'''
+        # checks that checkVarHGVS works for substitution in BRCA1
+        checkVar = checkVarHGVS.checkVarHGVS(self.varBRCA1)
+        self.assertTrue(checkVar)
+
+        self.varBRCA1["Ref"] = "T"
+        self.varBRCA1["Alt"] = "G"
+        checkVar = checkVarHGVS.checkVarHGVS(self.varBRCA1)
+        self.assertFalse(checkVar["Consistent"])
+        
+        # checks that checkVarHGVS works for substitution in BRCA2
+        checkVar = checkVarHGVS.checkVarHGVS(self.varBRCA2)
+        self.assertTrue(checkVar)
+
+        self.varBRCA2["Ref"] = "T"
+        self.varBRCA2["Alt"] = "G"
+        checkVar = checkVarHGVS.checkVarHGVS(self.varBRCA2)
+        self.assertFalse(checkVar["Consistent"])
+
+    def test_checkVarHGVSDelSingle(self):
+        '''Tests checkVarHGVS for single nucleotide deletion variants'''
+        # checks that checkVarHGVS works for single nucleotide deletion in BRCA1
+        self.varBRCA1["Ref"] = "GA"
+        self.varBRCA1["Alt"] = "G"
+        self.varBRCA1["pyhgvs_cDNA"] = "NM_007294.3:c.*179delT"
```

Comment:
this is exactly what im thinking for the above! even if it's repetitive it's far easier for other developers to read and maintain the tests when it's explicit like this